motionplan/motionPlanner.go

```diff
@@ -140,80 +154,56 @@ func motionPlanInternal(ctx context.Context,
 		return nil, err
 	}
 
-	opts := make([]map[string]interface{}, 0, len(goals))
-
-	// If no planning opts, use default. If one, use for all goals. If one per goal, use respective option. Otherwise error.
-	if len(motionConfigs) != len(goals) {
-		switch len(motionConfigs) {
-		case 0:
-			for range goals {
-				opts = append(opts, map[string]interface{}{})
-			}
-		case 1:
-			// If one config passed, use it for all waypoints
-			for range goals {
-				opts = append(opts, motionConfigs[0])
-			}
-		default:
-			return nil, errors.New("goals and motion configs had different lengths")
-		}
-	} else {
-		opts = motionConfigs
+	// Create a frame to solve for, and an IK solver with that frame.
+	sf, err := newSolverFrame(fs, solveFrameList, goal.Parent(), seedMap)
+	if err != nil {
+		return nil, err
 	}
-
-	// Each goal is a different PoseInFrame and so may have a different destination Frame. Since the motion can be solved from either end,
-	// each goal is solved independently.
-	for i, goal := range goals {
-		// Create a frame to solve for, and an IK solver with that frame.
-		sf, err := newSolverFrame(fs, solveFrameList, goal.Parent(), seedMap)
-		if err != nil {
-			return nil, err
-		}
-		if len(sf.DoF()) == 0 {
-			return nil, errors.New("solver frame has no degrees of freedom, cannot perform inverse kinematics")
-		}
-		seed, err := sf.mapToSlice(seedMap)
-		if err != nil {
-			return nil, err
-		}
-		startPose, err := sf.Transform(seed)
+	if len(sf.DoF()) == 0 {
+		return nil, errors.New("solver frame has no degrees of freedom, cannot perform inverse kinematics")
+	}
+	seed, err := sf.mapToSlice(seedMap)
+	if err != nil {
+		return nil, err
+	}
+	startPose, err := sf.Transform(seed)
+	if err != nil {
+		return nil, err
+	}
+	wsPb := &commonpb.WorldState{}
```

Comment:
looks like we're just converting to protobuf here to print right? We should long term just add a pretty print method to world state. Doesn't have to be on this PR though

Reply:
Agree, probably an improvement we can ticket. I'll put that on my list to file.

Reply:
That would be great!